addons/website_sale/controllers/main.py

```diff
@@ -667,10 +667,11 @@ def _prepare_product_values(self, product, category, search, **kwargs):
         product_markup_data = [product._to_markup_data(request.website)]
         if category:
             category = ProductCategory.browse(int(category)).exists()
-            # Add breadcrumb's SEO data.
-            product_markup_data.append(self._prepare_breadcrumb_markup_data(
-                request.website.get_base_url(), category, product.name
-            ))
```

Comment:
Please avoid to imbricate `if` when it's not needed.
```suggestion
        category = ProductCategory.browse(int(category)).exists()
        if category:
            # Add breadcrumb's SEO data.
            product_markup_data.append(self._prepare_breadcrumb_markup_data(
                request.website.get_base_url(), category, product.name
            ))
```

Reply:
Hey @chevalierv 
IMO we should do as below, preventing error when the category is None.
```<py
if category := ProductCategory.browse(category and int(category)).exists():
```

What do you think?

Reply:
Yes thanks for checking!